config/default.yml

```diff
@@ -1902,6 +1982,11 @@ Lint/LiteralAsCondition:
   Enabled: true
   VersionAdded: '0.51'
 
+Lint/LiteralAssignmentInCondition:
+  Description: 'Checks for literal assignments in the conditions.'
+  Enabled: pending
+  VersionAdded: '1.58'
```

Comment:
Unsure about this one. From time to time, I write this – and have seen it in the wild as well:
```
if (user = data.dig(0, :rel, :user))
  ...
```
Or will usage with parens still be allowed?

Reply:
From my understanding, this cop is only about literal assignments, i.e. `if (x = 1)`.

Reply:
Ah I see. Should be fine then.